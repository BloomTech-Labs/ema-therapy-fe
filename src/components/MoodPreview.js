import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Card from './Card';
import moodToString from '../utils/moodToString';
import happyPot from '../assets/happy200h.png';
import reallyHappyPot from '../assets/reallyhappy200h.png';
import normalPot from '../assets/normal200h.png';
import sadPot from '../assets/sad200h.png';
import unhappyPot from '../assets/unhappy200h.png';
import styles from '../styles/theme';
import formatDate from '../utils/formatDate';

function MoodPreview({ lastItem, count }) {
  return (
    <PreviewContainer>
      <DayWrapper>
        <p className="weekday">{formatDate(lastItem.createdAt, 'iii')}</p>
      </DayWrapper>
      <StyledMoodCard lastItem={lastItem}>
        <div>
          <p className="time">{formatDate(lastItem.createdAt, 'h:mm a')}</p>
          <p className="mood">{moodToString(lastItem.mood)}</p>
          <p className="count">{count > 1 ? `${count} entries` : '1 entry'}</p>
        </div>
      </StyledMoodCard>
    </PreviewContainer>
  );
}

MoodPreview.propTypes = {
  lastItem: PropTypes.shape({
    id: PropTypes.string.isRequired,
    createdAt: PropTypes.string.isRequired,
    mood: PropTypes.number.isRequired,
  }).isRequired,
  count: PropTypes.number.isRequired,
};

export default MoodPreview;

const PreviewContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 23px;

  .weekday {
    color: #0c423b;
    font-size: 12px;
    width: 30px;
    margin: 0 6px 0 0;
  }
`;

const DayWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

// conditionally render the bg img path based on props
function getPath(p) {
  let path;
  if (p.lastItem.mood === 1) path = unhappyPot;
  if (p.lastItem.mood === 2) path = sadPot;
  if (p.lastItem.mood === 3) path = normalPot;
  if (p.lastItem.mood === 4) path = happyPot;
  if (p.lastItem.mood === 5) path = reallyHappyPot;
  return path;
}

const StyledMoodCard = styled(Card)`
  padding: 7px 16px 8px;
  height: 95px;
  width: 100%;
  background-image: url(${getPath});
  background-repeat: no-repeat;
  background-position: top -18px right -22px;
  background-size: 115px 135px;
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.15);

  .time {
    margin: 0;
    font-size: 11px;
    color: ${styles.brightYellow};
  }

  .mood {
    font-size: 16px;
    margin: 8px 0 15px;
    font-weight: 500;
    text-transform: capitalize;
    color: #0c423b;
  }

  .count {
    color: #fca395;
    font-style: italic;
    font-size: 10px;
    margin-bottom: 10px;
    text-align: end;
    margin-right: 95px;
  }
`;
