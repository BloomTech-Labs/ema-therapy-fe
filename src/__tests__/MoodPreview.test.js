import React from 'react';
import { render } from '@testing-library/react';
import MoodPreview from '../components/MoodPreview';

const mockMoodPreview = (
  <MoodPreview
    lastItem={{
      mood: 1,
      id: 'mockId',
      sleep: 1,
      anxietyLevel: 1,
      text: 'mockText',
      createdAt: '1575270000000', // Monday, December 2, 2019 1:00:00 AM (CST)
      weather: 'mockWeather',
    }}
    count={1}
  />
);
describe('MoodPreview component', () => {
  test('renders without crashing', () => {
    render(mockMoodPreview);
  });

  test('matches snapshot', () => {
    expect(render(mockMoodPreview)).toMatchSnapshot();
  });
});
