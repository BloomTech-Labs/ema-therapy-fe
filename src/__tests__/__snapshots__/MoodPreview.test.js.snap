// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodPreview component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bwzfXH gLhJZY"
      >
        <div
          class="sc-htpNat chPvqm"
        >
          <p
            class="weekday"
          >
            Mon
          </p>
        </div>
        <div
          class="sc-bdVaJa fcqkUA sc-bxivhb chPzur"
        >
          <div>
            <p
              class="time"
            >
              1:00 AM
            </p>
            <p
              class="mood"
            >
              unhappy
            </p>
            <p
              class="count"
            >
              1 entry
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bwzfXH gLhJZY"
    >
      <div
        class="sc-htpNat chPvqm"
      >
        <p
          class="weekday"
        >
          Mon
        </p>
      </div>
      <div
        class="sc-bdVaJa fcqkUA sc-bxivhb chPzur"
      >
        <div>
          <p
            class="time"
          >
            1:00 AM
          </p>
          <p
            class="mood"
          >
            unhappy
          </p>
          <p
            class="count"
          >
            1 entry
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
